engine/health_check.go

```diff
@@ -123,10 +123,11 @@ func checkHTTP(ID string, backends []string, code int, timeout time.Duration) bo
 func checkTCP(ID string, backends []string, timeout time.Duration) bool {
 	for _, backend := range backends {
 		log.Debugf("[checkTCP] Check health via tcp: container %s, backend %s", ID, backend)
-		_, err := net.DialTimeout("tcp", backend, timeout)
+		conn, err := net.DialTimeout("tcp", backend, timeout)
```

Comment:
Actually, GC will handle this automatically, using `runtime.SetFinalizer()`, no bother to close the connection manually since there are not many ports to check, but if the user sets a high check frequency, close the connection manually can save some resources. However, the application should set a `ReadTimeout` or `WriteTimeout` for the connection, so the health check conn can be closed in the server side.

Reply:
For some special service it's correct. But for some protocol based on text-like protocol like redis, the established connections count can be over 80.

I noticed the situation and checked why there are so many connections after migrating to container and finally found this. So i add this to open and close the resource in time just as what  i did in c or java. And i think maybe we should not make it in reliance on GC, should we?

Reply:
Yep, close the connection is good.